cmd/rpcdaemon/commands/graphql_api.go

```diff
@@ -0,0 +1,171 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/holiman/uint256"
+	"github.com/ledgerwatch/erigon-lib/chain"
+	"github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/common/hexutil"
+	"github.com/ledgerwatch/erigon/consensus/ethash"
+	"github.com/ledgerwatch/erigon/core/rawdb"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/rpc"
+	"github.com/ledgerwatch/erigon/turbo/adapter/ethapi"
+	"github.com/ledgerwatch/erigon/turbo/rpchelper"
+)
+
+type GraphQLAPI interface {
+	GetBlockDetails(ctx context.Context, number rpc.BlockNumber) (map[string]interface{}, error)
+	// GetBlockDetailsByHash(ctx context.Context, hash common.Hash) (map[string]interface{}, error)
+	// GetBlockTransactions(ctx context.Context, number rpc.BlockNumber, pageNumber uint8, pageSize uint8) (map[string]interface{}, error)
+}
+
+type GraphQLAPIImpl struct {
+	*BaseAPI
+	db kv.RoDB
+}
+
+func NewGraphQLAPI(base *BaseAPI, db kv.RoDB) *GraphQLAPIImpl {
+	return &GraphQLAPIImpl{
+		BaseAPI: base,
+		db:      db,
+	}
+}
+
+func (api *GraphQLAPIImpl) GetBlockDetails(ctx context.Context, number rpc.BlockNumber) (map[string]interface{}, error) {
+	tx, err := api.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	b, senders, err := api.getBlockWithSenders(ctx, number, tx)
+	if err != nil {
+		return nil, err
+	}
+	if b == nil {
+		return nil, nil
+	}
+
+	chainConfig, err := api.chainConfig(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	receipts, err := api.getReceipts(ctx, tx, chainConfig, b, b.Body().SendersFromTxs())
+	if err != nil {
+		return nil, fmt.Errorf("getReceipts error: %w", err)
+	}
+	result := make([]map[string]interface{}, 0, len(receipts))
+	for _, receipt := range receipts {
+		txn := b.Transactions()[receipt.TransactionIndex]
+		result = append(result, marshalReceipt(receipt, txn, chainConfig, b.HeaderNoCopy(), txn.Hash(), true))
+	}
+
+	getBlockRes, err := api.delegateGetBlockByNumber(tx, b, number, false)
+	if err != nil {
+		return nil, err
+	}
+	getIssuanceRes, err := api.delegateIssuance(tx, b, chainConfig)
+	if err != nil {
+		return nil, err
+	}
+	feesRes, err := api.delegateBlockFees(ctx, tx, b, senders, chainConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	response := map[string]interface{}{}
+	response["block"] = getBlockRes
+	response["issuance"] = getIssuanceRes
+	response["totalFees"] = hexutil.Uint64(feesRes)
+	response["receipts"] = result
+	return response, nil
+}
+
+func (api *GraphQLAPIImpl) getBlockWithSenders(ctx context.Context, number rpc.BlockNumber, tx kv.Tx) (*types.Block, []common.Address, error) {
```

Comment:
BaseApi has similar method already? (It also using lru cache as i remember)

Reply:
I copied that from OTS code, would try to find BaseApi better method